sycl/source/detail/scheduler/graph_processor.cpp

```diff
@@ -59,6 +59,10 @@ bool Scheduler::GraphProcessor::handleBlockingCmd(Command *Cmd,
       const EventImplPtr &RootCmdEvent = RootCommand->getEvent();
       Cmd->addBlockedUserUnique(RootCmdEvent);
       EnqueueResult = EnqueueResultT(EnqueueResultT::SyclEnqueueBlocked, Cmd);
+
+      // Blocked command will be enqueued asynchronously from submission so we
+      // need to keep current root code location to report failure properly.
+      RootCommand->copySubmissionCodeLocation();
```

Comment:
Are we trying to copy the TLS code location info to RootCommand? Shouldn't it already be there if we had it in the constructor?

Reply:
we keep it in the constructor for host task only. Although if kernel depends on host task - it will be blocked by it and enqueued in host task thread after host kernel completion. To handle it I copy code location here.